app/controllers/ExperimentsController.php

```diff
@@ -143,11 +143,15 @@
         try {
             $Entity->setId($_POST['id']);
             $Entity->canOrExplode('write');
-            $ts = new TrustedTimestamps(new Config(), new Teams($_SESSION['team_id']), $Entity);
-            if ($ts->timeStamp()) {
-                echo json_encode(array(
+            if ($Entity->isTimestampable()) {
+                $ts = new TrustedTimestamps(new Config(), new Teams($_SESSION['team_id']), $Entity);
+                if ($ts->timeStamp()) {
+                    echo json_encode(array(
                     'res' => true
                 ));
+                }
+            } else {
+                throw new Exception(_('This experiment cannot be timestamped!'));
```

Comment:
Ok, I can do that. But if you comment out the code that is hiding the timestamp button and try to timestamp, the message from the exception is picked up just fine by the usual note element.